common/components/box_collider.py

```diff
@@ -0,0 +1,35 @@
+from __future__ import annotations
+
+from common.engine import component
+from common.utils.vector2 import Vector2
+
+
+@component
+class BoxCollider:
+    __size: Vector2
+
+    @classmethod
+    def from_size(cls, x: int, y: int) -> BoxCollider:
+        return cls(Vector2(x, y))
+
+    def __init__(self, vector: Vector2) -> None:
+        self.__size = vector
+
+    @property
+    def x(self) -> float:
```

Comment:
`from_size` uses `int`, while the property return `float`
Maybe `Vector` should be generics (int, float) btw